File: R/demo/app-chat.R
library(shiny)
library(bslib)
library(ellmer)
library(shinychat)

ui <- page_fluid(
  theme = bs_theme(bootswatch = "minty"),

  # App title
  h1("DeckCheck"),

  # Create a card
  card(
    card_header("Get started"),
    p("Ask me anything about your presentation 💡"),

    # Chat component
    chat_mod_ui("chat_component")
  )
)

server <- function(input, output, session) {
  chat <- chat_anthropic(
    model = "claude-sonnet-4-20250514",
    system_prompt = "You are a presentation coach for data scientists. 
  You give constructive, focused, and practical feedback on titles, structure, and storytelling."
  )

  chat_mod_server("chat_component", chat)
}

shinyApp(ui, server)

File: Py/demo/app-chat.py
from dotenv import load_dotenv
from shiny import App, ui
from chatlas import ChatAnthropic

load_dotenv()  # Loads key from the .env file

# Define UI
app_ui = ui.page_fluid(
    ui.h1("DeckCheck"),
    # Card with chat component
    ui.card(
        ui.card_header("Get started"),
        ui.p("Ask me anything about your presentation 💡"),
        # Chat component
        ui.chat_ui(id="my_chat"),
    ),
)


# Define server
def server(input, output, session):
    chat_component = ui.Chat(id="my_chat")

    chat = ChatAnthropic(
        model="claude-sonnet-4-20250514",
        system_prompt="You are a presentation coach for data scientists. You give constructive, focused, and practical feedback on titles, structure, and storytelling.",
    )

    @chat_component.on_user_submit
    async def handle_user_input(user_input: str):
        response = await chat.stream_async(user_input)
        await chat_component.append_message_stream(response)


# Create app
app = App(app_ui, server)
